packages/hint-compat-api/src/helpers/compat-api.ts

```diff
@@ -0,0 +1,156 @@
+            return undefined;
+        }
+
+        // Sometimes the API give an array but only the first seems relevant
+        if (Array.isArray(currentBrowserFeatureSupported) && currentBrowserFeatureSupported.length > 0) {
+            if (prefix) {
+                currentBrowserFeatureSupported = currentBrowserFeatureSupported.find((info) => {
+                    return info.prefix === prefix;
+                });
+            } else {
+                currentBrowserFeatureSupported = currentBrowserFeatureSupported.find((info) => {
+                    return !info.prefix;
+                });
+            }
+        }
+
+        return currentBrowserFeatureSupported as SimpleSupportStatement;
+    }
+
+    /* eslint-disable camelcase */
+    public getWorstCaseSupportStatementFromInfo(browserFeatureSupported: SupportStatement | undefined): SimpleSupportStatement | undefined {
+        // If we dont have information about the compatibility, ignore.
+        if (!browserFeatureSupported) {
+            return browserFeatureSupported;
+        }
+
+        // Take the smaller version_removed and bigger version_added
+        const worstBrowserFeatureSupported: SimpleSupportStatement = {
+            version_added: null,
+            version_removed: null
+        };
+
+        if (Array.isArray(browserFeatureSupported) && browserFeatureSupported.length > 0) {
+            browserFeatureSupported.forEach((info) => {
+                if (!worstBrowserFeatureSupported.version_added && info.version_added === true) {
+                    worstBrowserFeatureSupported.version_added = true;
+                }
+
+                if (!worstBrowserFeatureSupported.version_added || worstBrowserFeatureSupported.version_added && info.version_added && info.version_added > worstBrowserFeatureSupported.version_added) {
+                    worstBrowserFeatureSupported.version_added = info.version_added;
+                }
+
+                if (!worstBrowserFeatureSupported.version_removed && info.version_removed === true) {
+                    worstBrowserFeatureSupported.version_removed = true;
+                }
+
+                if (!worstBrowserFeatureSupported.version_removed || worstBrowserFeatureSupported.version_removed && info.version_removed && info.version_removed < worstBrowserFeatureSupported.version_removed) {
+                    worstBrowserFeatureSupported.version_removed = info.version_removed;
+                }
+            });
+
+            return worstBrowserFeatureSupported;
+        }
+
+        return browserFeatureSupported as SimpleSupportStatement;
+    }
+    /* eslint-enable camelcase */
+
+    private isFeatureRequiredToTest(typedFeatureValue: CompatStatement & MDNTreeFilteredByBrowsers, isCheckingNotBroadlySupported = false): boolean {
+        // TODO: Here we are checking only parent but this object has children
+        let isRequiredToTest = false;
+
+        forEach(this.browsers, (browserVersionsList, browser) => {
```

Comment:
If I've understood the code correctly, this could be improved by doing [`Array.prototype.some()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) so the loop stops the moment there's a `true` value returned.

Reply:
Yes! nice feedback, we will use _.some to be easier for collections if this is ok

Reply:
I'd rather us use the builtin methods. So far the only methods used from Lodash are `forEach` and now `some` too which IMO doesn't justify using it and pulling another package.